packages/chain-agnostic-permission/src/adapters/caip-permission-adapter-accounts.ts

```diff
@@ -139,3 +140,99 @@ export const setEthAccounts = (
     ),
   };
 };
+
+/**
+ * Sets the permitted accounts for the given scopes object.
+ *
+ * @param scopesObject - The scopes object to set the permitted accounts for.
+ * @param accounts - The permitted accounts to add to the appropriate scopes.
+ * @returns The updated scopes object with the permitted accounts set.
+ */
+const setPermittedAccountsForScopesObject = (
+  scopesObject: InternalScopesObject,
+  accounts: CaipAccountId[],
+) => {
+  const updatedScopesObject: InternalScopesObject = {};
+  Object.entries(scopesObject).forEach(([key, scopeObject]) => {
+    // Cast needed because index type is returned as `string` by `Object.entries`
+    const scopeString = key as keyof typeof scopesObject;
+    const { namespace, reference } = parseScopeString(scopeString);
+
+    let caipAccounts: CaipAccountId[] = [];
+    if (namespace && reference) {
+      caipAccounts = accounts.reduce<CaipAccountId[]>((acc, account) => {
+        const {
+          chain: { namespace: accountNamespace },
+          address: accountAddress,
+        } = parseCaipAccountId(account);
+        // If the account namespace is the same as the scope namespace, add the account to the scope
+        // This will, for example, distribute all EIP155 accounts, regardless of reference, to all EIP155 scopes
+        if (namespace === accountNamespace) {
+          acc.push(`${namespace}:${reference}:${accountAddress}`);
+        }
+        return acc;
+      }, []);
+    }
+
+    updatedScopesObject[scopeString] = {
+      ...scopeObject,
+      accounts: uniq(caipAccounts),
+    };
+  });
+
+  return updatedScopesObject;
+};
+
+/**
+ * Sets the permitted accounts for the given CAIP-25 caveat value.
+ *
+ * @param caip25CaveatValue - The CAIP-25 caveat value to set the permitted accounts for.
+ * @param accounts - The permitted accounts to add to the appropriate scopes.
+ * @returns The updated CAIP-25 caveat value with the permitted accounts set.
+ */
+export const setPermittedAccounts = (
+  caip25CaveatValue: Caip25CaveatValue,
+  accounts: CaipAccountId[],
+): Caip25CaveatValue => {
+  return {
+    ...caip25CaveatValue,
+    requiredScopes: setPermittedAccountsForScopesObject(
+      caip25CaveatValue.requiredScopes,
+      accounts,
+    ),
+    optionalScopes: setPermittedAccountsForScopesObject(
+      caip25CaveatValue.optionalScopes,
+      accounts,
+    ),
+  };
+};
+
+/**
+ * Gets the requested accounts from the given CAIP-25 caveat value.
+ *
+ * @param requestedCaip25CaveatValue - CAIP-25 request values.
+ * @returns Accounts available for requesting.
+ */
+export function getAllAccountsFromCaip25Caveat(
```

Comment:
thoughts on just calling this `getPermittedAccounts`? The ending of this one sticks out from the rest. I can see why `getPermittedAccounts` might conflict in other contexts though, but maybe that can be fixed with a rename on import?

Reply:
Agreed that its not a great name, especially since others don't follow that naming convention. I'm going to change it to `getAllAccounts` for now since really this is used before the caveat is persisted... but it can be used for the caveat persisted or not...